Nesteo.Server.DataImport/Program.cs

```diff
@@ -0,0 +1,420 @@
+
+            [Index(8)]
+            public string Material { get; set; }
+
+            [Index(9)]
+            public string Loch { get; set; }
+
+            [Index(10)]
+            public string Bemerkungen { get; set; }
+
+            [Index(11)]
+            public string AktualisiertDatum { get; set; }
+        }
+
+        public class Kontrolldaten
+        {
+            [Index(0)]
+            public string NistkastenNummer { get; set; }
+
+            [Index(1)]
+            public string Datum { get; set; }
+
+            [Index(2)]
+            public string ZustandKasten { get; set; }
+
+            [Index(3)]
+            public string Gereinigt { get; set; }
+
+            [Index (4)]
+            public string Besetzt { get; set; }
+
+            [Index(5)]
+            public string AnzahlEier { get; set; }
+
+            [Index(6)]
+            public string AnzahlJungvogel { get; set; }
+
+            [Index(7)]
+            public string AlterJungvogel { get; set; }
+
+            [Index(8)]
+            public string Vogelart { get; set; }
+
+            [Index(9)]
+            public string Berignt { get; set; }
+
+            [Index(10)]
+            public string Bemerkungen { get; set; }
+        }
+
+        public static async Task Main(string[] args)
+        {
+
+            // Create host
+            IHost host = Server.Program.CreateHostBuilder(args).Build();
+            await Server.Program.PrepareHostAsync(host).ConfigureAwait(false);
+
+            // Request database context
+            NesteoDbContext dbContext = host.Services.GetRequiredService<NesteoDbContext>();
+
+            // Safety check
+//            Console.Write("Do you want to fill the database with sample data? Please type 'yes': ");
+//            if (Console.ReadLine()?.ToLower() != "yes")
+//                return;
+//            Console.Write("Do you really want to do this? This will DELETE all existing data! Please type 'yes': ");
+//            if (Console.ReadLine()?.ToLower() != "yes")
+//                return;
+
+            // Clear database
+//            await ClearDatabase(dbContext);
+
+            UserEntity user = await dbContext.Users.FirstOrDefaultAsync().ConfigureAwait(false);
+            if (user == null)
+            {
+                Console.WriteLine("Please run the server first to make sure that at least one user exists.");
+                return;
+            }
+
+            ReadStammDaten(dbContext, user);
+            await SaveDatabase(dbContext);
+
+            ReadKontrollDaten(dbContext, user);
+            await SaveDatabase(dbContext);
+        }
+
+        private static async Task SaveDatabase(NesteoDbContext dbContext)
+        {
+            Console.WriteLine("Saving");
+            await dbContext.SaveChangesAsync().ConfigureAwait(false);
+            Console.WriteLine("Done.");
+        }
+
+        private static async void ReadStammDaten(NesteoDbContext dbContext, UserEntity user)
```

Comment:
Ah, what you probably didn't know yet: `async void` is really bad and could break exception handling when not being very careful. Always use `async Task` please.